SNMP exporter: Use a "name" argument for "target" and "walk_param" blocks (#1790)
grafana/alloy: docs/sources/reference/components/prometheus/prometheus.exporter.snmp.md

### target block

The `target` block defines an individual SNMP target.
The `target` block may be specified multiple times to define multiple targets.

| Name           | Type     | Description                                                           | Default | Required |
| -------------- | -------- | --------------------------------------------------------------------- | ------- | -------- |
| `name`         | `string` | The name of the target.                                               |         | yes      |
| `address`      | `string` | The address of SNMP device.                                           |         | yes      |
| `module`       | `string` | SNMP module to use for polling.                                       | `""`    | no       |
| `auth`         | `string` | SNMP authentication profile to use.                                   | `""`    | no       |
| `walk_params`  | `string` | Config to use for this target.                                        | `""`    | no       |
| `snmp_context` | `string` | Override the `context_name` parameter in the SNMP configuration file. | `""`    | no       |

> The value of the `name` argument will be set as the value of the target's `job` label.

Reviewer: ```suggestion
The value of the `name` argument is set to the value of the target's `job` label.
```

{{< admonition type="note" >}}

> Instead of a `name` argument, previous versions of Alloy used to use a block label.

Reviewer: ```suggestion
Instead of a `name` argument, previous versions of {{< param "PRODUCT_NAME" >}} used a block label.
```

> See the [Deprecated features](#deprecated-features) section for more information.

Reviewer: ```suggestion
Refer to [Deprecated features](#deprecated-features) for more information.
```

{{< /admonition >}}

> If `name` is not set, the target's `job` label will be set to the label of the `target` block.

Reviewer: ```suggestion
If `name` is not set, the target's `job` label is set to the label of the `target` block.
```

### walk_param block

The `walk_param` block defines an individual SNMP connection profile that can be used to override default SNMP settings.
The `walk_param` block may be specified multiple times to define multiple SNMP connection profiles.

| Name              | Type       | Description                                   | Default | Required |
| ----------------- | ---------- | --------------------------------------------- | ------- | -------- |
| `name`            | `string`   | Name of the module to override.               |         | yes      |
| `max_repetitions` | `int`      | How many objects to request with GET/GETBULK. | `25`    | no       |
| `retries`         | `int`      | How many times to retry a failed request.     | `3`     | no       |
| `timeout`         | `duration` | Timeout for each individual SNMP request.     |         | no       |

{{< admonition type="note" >}}

> Instead of a `name` argument, previous versions of Alloy used to use a block label.

Reviewer: ```suggestion
Instead of a `name` argument, previous versions of {{< param "PRODUCT_NAME" >}} used a block label.
```

> See the [Deprecated features](#deprecated-features) section for more information.

Reviewer: ```suggestion
Refer to [Deprecated features](#deprecated-features) for more information.
```

`prometheus.exporter.snmp` does not expose any component-specific
debug metrics.

## Deprecated features

> In previous versions of Alloy, the names `target` blocks and `walk_param` blocks were specified via a block label:

Reviewer: ```suggestion
In previous versions of {{< param "PRODUCT_NAME" >}}, the names `target` blocks and `walk_param` blocks were specified via a block label:
```

```alloy
target "example_name_of_target" {
    ...
}

walk_param "example_name_of_walk_param" {
    ...
}
```

> Users are now advised to switch to using a `name` attribute instead:

Reviewer: ```suggestion
You can switch to using a `name` attribute instead of using a block label:
```

```alloy
target {
    name = "example_name_of_target"
    ...
}

walk_param {
    name = "example_name_of_walk_param"
    ...
}
```

> Block labels for `target` and `walk_param` are deprecated and will be removed in version 2.0 of Alloy.
> In Alloy version 1, it will remain possible to use a block label instead of the `name` attribute.
> However, the `name` attribute is documented as "required" to encourage users to use it and to remove confusion.

Reviewer: ```suggestion
Block labels for `target` and `walk_param` are deprecated and will be removed in {{< param "PRODUCT_NAME" >}} 2.0.
In {{< param "PRODUCT_NAME" >}} 1.x, you can continue to use a block label.
However, the `name` attribute is documented as "required" to encourage users to use it and to remove confusion.
```

What happens if the user keeps the block labels AND adds a name attribute?

> The benefits of using a `name` attribute include:
> * Not being subject to restrictions on [Alloy Syntax Identifiers][syntax-identifiers].
> * It is possible to get the value of a `name` argument from a component such as [local.file][] or [remote.http][].

Reviewer: ```suggestion
The benefits of using a `name` attribute include:
* You aren't subject to restrictions on [Alloy Syntax Identifiers][syntax-identifiers].
* You can get the value of a `name` argument from a component such as [local.file][] or [remote.http][].
```
